onchain/src/contracts/account.cairo

```diff
@@ -86,13 +93,49 @@ mod Account {
 
     #[constructor]
     fn constructor(
-        ref self: ContractState, public_key: felt252, approver: ContractAddress, limit: u256
+        ref self: ContractState, public_key: felt252, approver: ContractAddress, limit: u256,
     ) {
         self.account.initializer(:public_key);
         self.transaction_approval.initializer(:approver);
         self.weekly_limit.initializer(:limit);
     }
 
+    #[derive(Serde, Drop)]
+    struct Claim {
+        amount: u256,
+        nonce: felt252,
+        signature: Array<felt252>
+    }
+    #[starknet::interface]
+    pub trait ClaimLinkTrait<T> {
+        fn claim(ref self: T, claim: Claim);
+        #[cfg(test)]
+        fn set_usdc_address(ref self: T, usdc_address: ContractAddress);
+    }
+    #[abi(embed_v0)]
+    impl ClaimLink of ClaimLinkTrait<ContractState> {
+        fn claim(ref self: ContractState, claim: Claim) {
```

Comment:
* i didn't know it existed
* very complex for such a simple use case

Reply:
I agree that it's also a bit overkill given that the original purpose of the EIP-712 was to provide wallets a standard to display signatures informations, which is useless for Vault. But there's still some benefits to it:

- An [OZ implementation](https://github.com/OpenZeppelin/cairo-contracts/blob/main/src/utils/cryptography/snip12.cairo)
- You specify the chain ID of your app in the message you sign, so your testnet signatures cannot be replayed
- There's also the version of the consuming protocol in the message, so you can invalidate every signatures in a simple protocol upgrade
- And it may not be the case for the swift package, but there's a few implementations of the offchain signing logic so you can easily implement it on the frontend

Reply:
ok you convinced me